control-station/src/services/PodSocketClient.ts

```diff
@@ -114,6 +120,25 @@ class PodSocketClient {
 	private onData(data: string): void {
 		console.log("server says", data);
 	}
+
+	private addMessage(response: string, newState: State): void {
+		const currentTime = new Date();
+		const hours = currentTime.getHours().toString().padStart(2, "0");
+		const minutes = currentTime.getMinutes().toString().padStart(2, "0");
+		const seconds = currentTime.getSeconds().toString().padStart(2, "0");
+		const timestamp = `${hours}:${minutes}:${seconds}`;
+
+		const newMessage: Message = {
+			timestamp: timestamp,
+			message: response,
+		};
+
+		this.setPodData((d) => ({
+			...d,
+			state: newState,
+			messages: d.messages ? [...d.messages, newMessage] : [newMessage],
```

Comment:
Question: is there a particular reason you needed to use a ternary here? Spreading an empty array should still work.
```suggestion
			messages: [...d.messages, newMessage],
```